Answer the questions by modifying the lines that says "REPLACE THIS LINE WITH YOUR ANSWER TO QUESTION XX".

**DO NOT MODIFY THE OTHER LINES. CHANGING OTHER LINES MAKE GRADING HARDER AND CAN REDUCE YOUR SCORE.**

**EACH OF YOUR ANSWER MUST BE AN ASSIGNMENT TO THE CORRESPONDING VARIABLE ans_qX (X for 1,2,...)**

We use pizza price dataset https://www.kaggle.com/datasets/timmofeyy/-pizza-prices-menu-in-13-different-pizzerias

In [1]:
import numpy as np
import pandas as pd
import matplotlib
url = 'https://www.dropbox.com/s/g0t2dcud76mykzt/pizza13_sample.csv?dl=1'
df = pd.read_csv(url)
df = df.drop(columns = ["Unnamed: 0", "Unnamed: 0.1"])
submission_mode = "s3"

#### Columns

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1347 entries, 0 to 1346
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   food    1334 non-null   object 
 1   size    1123 non-null   object 
 2   price   1333 non-null   float64
 3   DBA     1347 non-null   object 
dtypes: float64(1), object(3)
memory usage: 42.2+ KB


### QUESTIONS Start here


Score for each question: 10 pts for Q1, 15 pts for each other questions.

Q1. Create an **int** variable **ans_q1** that describes the number of rows in this dataframe.

In [3]:
ans_q1 = int(df.shape[0])
ans_q1

1347

Sample answer

1347


Q2. Create an **int** variable **ans_q2** that describes the number of pizzerias (i.e., number of unique DBAs) in this dataframe.

In [4]:
# REPLACE THIS LINE WITH YOUR ANSWER TO QUESTION 2
ans_q2 = df['DBA'].nunique()
ans_q2

13

Sample answer

13


Q3. Create a **dataframe** variable **ans_q3** that describes all the menus of Dominos

In [5]:
# REPLACE THIS LINE WITH YOUR ANSWER TO QUESTION 3
ans_q3 = df[df['DBA'] == 'Dominos']
ans_q3

food          size  price      DBA
0                                   Hand Tossed   Small (10")   5.99  Dominos
1                                   Hand Tossed  Medium (12")   7.99  Dominos
2                                   Hand Tossed   Large (14")   9.99  Dominos
3                                  Handmade Pan  Medium (12")   7.99  Dominos
4                            Crunchy Thin Crust   Small (10")   5.99  Dominos
..                                          ...           ...    ...      ...
139                     Golden Italian Dressing           Bag   0.59  Dominos
140                                Garden Ranch           Bag   0.59  Dominos
141  Fat-Free Raspberry or Balsamic Vinaigrette           Bag   0.59  Dominos
142                 Red Pepper Seasoning Flakes           NaN   0.99  Dominos
143                      Grated Parmesan Cheese           NaN   0.99  Dominos

[144 rows x 4 columns]

Sample answer

|index|food|size|price|DBA|
|---|---|---|---|---|
|0|Hand Tossed|Small \(10")|5\.99|Dominos|
|1|Hand Tossed|Medium \(12")|7\.99|Dominos|
|2|Hand Tossed|Large \(14")|9\.99|Dominos|
...
|141|Fat-Free Raspberry or Balsamic Vinaigrette|Bag|0\.59|Dominos|
|142|Red Pepper Seasoning Flakes|NaN|0\.99|Dominos|
|143|Grated Parmesan Cheese|NaN|0\.99|Dominos|


Q4. Create a **float** variable **ans_q4** that describes the average price of Dominos medium (12'') size pizza

In [6]:
# REPLACE THIS LINE WITH YOUR ANSWER TO QUESTION 4

ans_q4 = df[df['size'] == 'Medium (12")'].sum()['price'] /  (df['size'] == 'Medium (12")').sum()
ans_q4

13.5908

Sample answer

Should be a float variable around 13 (exact answer not disclosed).

Q. Create a **list** variable **ans_q5** that describes all menus in Peter Piper. Here, we only count the same food of different size once. 

In [7]:
# REPLACE THIS LINE WITH YOUR ANSWER TO QUESTION 5
ans_q5 = []
for i in df['food']:
    if i not in ans_q5:
        ans_q5.append(i)
ans_q5

['Hand Tossed',
 'Handmade Pan',
 'Crunchy Thin Crust',
 'Gluten Free Crust',
 'Spinach & Feta (Hand Tossed or Thin Crust)',
 'Spinach & Feta (Hand Tossed, Handmade Pan or Thin Crust)',
 'Spinach & Feta (Hand Tossed, Thin Crust or Brooklyn Style)',
 'Spinach & Feta (Brooklyn Style)',
 'Spinach & Feta (Gluten Free Crust)',
 'Wisconsin 6 Cheese (Hand Tossed or Thin Crust)',
 'Wisconsin 6 Cheese (Hand Tossed, Handmade Pan or Thin Crust)',
 'Wisconsin 6 Cheese (Hand Tossed, Thin Crust or Brooklyn Style)',
 'Wisconsin 6 Cheese (Brooklyn Style)',
 'Wisconsin 6 Cheese (Gluten Free Crust)',
 'Philly Cheese Steak (Hand Tossed or Thin Crust)',
 'Philly Cheese Steak (Hand Tossed, Handmade Pan or Thin Crust)',
 'Philly Cheese Steak (Hand Tossed, Thin Crust or Brooklyn Style)',
 'Philly Cheese Steak (Brooklyn Style)',
 'Philly Cheese Steak (Gluten Free Crust)',
 'Pacific Veggie (Hand Tossed or Thin Crust)',
 'Pacific Veggie (Hand Tossed, Handmade Pan or Thin Crust)',
 'Pacific Veggie (Hand Tossed, 

Sample answer

```
['Side of Chicken',
 'Wings (Spicy or Mild Buffalo, Sweet BBQ or Plain)',
 'Cinnamon Crunch Dessert',
  ...
 'The Werx (Original, Pan or Ultra Thin)',
 'Hearty Hawaiian (Original, Pan or Ultra Thin)',
 'Chopped House Salad']
 ```



Q. Return **series** variable **ans_q6** that describes the number of menus (rows) for each pizzerias.



In [8]:
# REPLACE THIS LINE WITH YOUR ANSWER TO QUESTION 6
food = sorted(list(df['DBA'].unique()))

ans_q6 = []
print('DBA')
for i in food:
    print(i,'  ',(df['DBA'] == i).sum())


DBA
Blaze    21
Dominos    144
Foxs    52
Godfathers    184
Hungry Howies    138
Imo's    89
Papa John's    147
Peter Piper    61
Pie Five    25
Pizza Hut    173
Pizza Inn    50
Round Table    130
Toppers    133


Sample answer



```
DBA
Blaze             21
Dominos          157
...
Round Table      155
Toppers          133
Name: food, dtype: int64
```



Q. Create a **dataframe** variable **ans_q7** that describes menus in **Pizza Hut** pizza. This dataframe has exactly three columns "food", "size", and "normalized_price".
The **normalized_price** columns shows the price of a pizza divided by the area of it. 

Note: The area of an $X$-inch pizza is given as $X*X*3.14/4$. You may assume that Pizza Hut pizza is of size

*   6 inches (Personal)
*   12 inches (Medium)
*   14 inches (Large)

For non-pizza items, keep this column NaN. Float variables must match sample answer at least by precision 0.001 for each datapoint. Hint: Use apply


In [33]:
# REPLACE THIS LINE WITH YOUR ANSWER TO QUESTION 7
import warnings
warnings.filterwarnings('ignore')

df1 = df[df['DBA']=='Pizza Hut']
def f(x):
    if x == 'Large':
        return 14*14*np.pi/4
    elif x == 'Medium':
        return 12*12*np.pi/4
    elif x == 'Personal':
        return 6*6*np.pi/4
    else:
        return None

df1['normalized_price'] = df1['price']/df1['size'].apply(lambda x: f(x))
df1 = df1[['food','size','normalized_price']]
df1.reset_index()

index          food      size  normalized_price
0      144   Hand-Tossed    Medium          0.075068
1      145   Hand-Tossed     Large          0.068144
2      146     Pan Pizza  Personal          0.151728
3      147     Pan Pizza    Medium          0.075068
4      148     Pan Pizza     Large          0.068144
..     ...           ...       ...               ...
168    312    Diet Pepsi   2 Liter               NaN
169    313  Mountain Dew    Bottle               NaN
170    314  Mountain Dew   2 Liter               NaN
171    315      Aquafina    Bottle               NaN
172    316    Dr. Pepper   2 Liter               NaN

[173 rows x 4 columns]

Sample answer

|index|food|size|normalized\_price|
|---|---|---|---|
|144|Hand-Tossed|Medium|0\.07510615711252654|
|145|Hand-Tossed|Large|0\.0681788639022488|
|146|Pan Pizza|Personal|0\.15180467091295116|
...
|314|Mountain Dew|2 Liter|NaN|
|315|Aquafina|Bottle|NaN|
|316|Dr\. Pepper|2 Liter|NaN|